src/client/components/MealsList.js

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import { useMeals } from "../contexts/MealsContext";
+import { Link } from "react-router-dom";
+import MealBox from "./MealBox";
+
+const MealsList = () => {
+  const { meals, loading, error } = useMeals();
+
+  const newMeals = [...meals];
+
+  newMeals.map((meal) => {
+    if (meal.max_number_of_guests > meal.reserved || meal.reserved === null) {
+      meal.availability = true;
+    } else meal.availability = false;
+  });
+
+  return (
+    <div>
+      <ul style={{ listStyleType: "none" }}>
+        {loading ? (
+          <li>Loading...</li>
+        ) : error ? (
+          <li>{error}</li>
+        ) : (
+          newMeals.map((meal) => {
```

Comment:
This becomes a bit hard to read. I would suggest having a separate return statement for the loading and error states.

Reply:
What I mean here is something like this:

```js
const MealsList = () => {
  // ...
  if (loading) return <p>Loading...</p>;
  if (error) return <p>{error}</p>;

  // And now you can have a return for the "normal" render without the ternary operators.
}
```